### DISCUSSION TITLE
Contributor Call: 4 March 2023

### DISCUSSION THREAD

**Discussion by l3xander** [freesewing/freesewing#3552] [Support] [answered]
You are all invited to the FreeSewing Contributor Call.
Please leave a comment here or [on Discord](https://discord.freesewing.org/) to add things to the agenda.
This issue will be updated after the call with notes and to-dos.

### About
FreeSewing's contributor call is an informal hangout/chat/meeting that is open to all FreeSewing contributors and users alike.
The call is organized/hosted by @tangerineshark and @kaermorhan.
The call does not have a fixed structure or agenda items. Everyone is free to suggest agenda items, ask questions, or bring ideas.

**Details**
Location: https://discord.freesewing.org (meeting room: 🔊`voice chat`)
Date: 4 March 2023 **5pm** UTC, which can be found in your local time zone [here](https://savvytime.com/converter/utc-to-belgium-antwerpen-ca-los-angeles-il-chicago-ny-new-york-city-brazil-sao-paulo-united-kingdom-london-france-paris-south-africa-johannesburg-msd-india-mumbai-hkt-japan-tokyo-australia-sydney/mar-4-2023/5pm).
Or check the Events tab in Discord!
The meetings will always be at 6pm, Antwerp time.

### Agenda

- #3575 (migrate documentation to *measurements sets*)
- **Proposal**: [Remove rigid documenation structure for pattern docs](https://github.com/freesewing/freesewing/discussions/3552#discussioncomment-5111285)
- **Proposal**: [Hide things differently](https://github.com/freesewing/freesewing/discussions/3586)
- **Proposal**: [Display cutting instructions on parts](https://github.com/freesewing/freesewing/discussions/3552#discussioncomment-5171267)
- **Proposal**: [the *fuck trees* setting](https://github.com/freesewing/freesewing/discussions/3552#discussioncomment-5171267)

**Comment by BenJamesBen** [comment]
"A more free-form format for design documentation, for v3"

In v2, designs specifically have 4 designer-authored documentation pages: 1. Cutting Instructions, 2. Fabric Options, 3. Sewing Instructions, 4. What You Need.
- Each of these 4 pages is required (the links are hardcoded in the design documentation index page).
- Only these 4 pages are allowed (the hardcoded index page doesn't allow for other pages).
- The 4 pages are hardcoded to be in that specific order.

(This is what the v3 design documentation looks like.)
![Screenshot 2023-02-25 at 2 53 44 PM](https://user-images.githubusercontent.com/109869956/221383321-090780c5-8ef0-41f5-822e-4ce18cfc5bb2.png)

For v3, I would like to see a more free-form approach to these design documentation pages. 
- Instead of the 4 hardcoded pages, there would be an arbitrary number of pages, with arbitrary titles. So for example, if the designer/author wanted to omit the Fabric Options page, they could do so. Or, if they wanted to include a "Historical Background" page, they could do so.
- The designer/author would be able to specify the order in which the documentation pages should appear (using `order:` FrontMatter in the documents). So for example, the designer/author could specify that "Historical Background" be the first documentation page listed.
- Any new approach that we implement would accommodate existing documentation for existing designs so they continue to work as-is, without needing to be modified. And, for new designs if the designer/author wanted to stick with the current 4-page format, they would be free to do so.

(The v3 approach would also need to allow us to address Bug https://github.com/freesewing/freesewing/issues/2025 "In the side bar, the pattern documentation sub pages are not listed in the same order as on the pattern documentation page".)

The new appoach would also give authors the ability to address Bug https://github.com/freesewing/freesewing/issues/2024 "Add additional information about each Pattern to that Pattern's documentation", if they choose to do so.

**Reply by joostdecock** [reply]
Added to the agenda

**Comment by eriese** [comment]
#3586 

And if there's time, I'd like to poll folks on how useful they would find it to have cutting instructions (e.g. `Cut 2 on fold from main fabric`) appended to the Title of parts, because there are a few of ways to handle it:

1. I make the title macro do something convoluted so that it actually runs after drafting, thereby making sure that cut information has been registered by the time the macro runs. This option allows designers to not have to make any changes to their designs with regard to titling
2. Designers make sure to call the title macro after adding their cut information, even if it was previously inheriting a title from a dependency
3. A separate macro for adding the cutting instruction text, which again would need to be called after final cutting information was added

And if there's still time, I'd like to discuss adding a `saveSpace` setting. Previously proposed as `fuckTrees`, this setting would be used during drafting to see if shortened pattern pieces should be drawn at full size, and if parts drawn from other parts (e.g. facings) should be drafted separately. My thinking is that this option will be helpful for cutting layout, where we need every piece in full in order to determine fabric usage.

**Comment by KaerMorhan** [comment, marked as answer]
**Notes**

Participants:

- @tangerineshark
- @lexander2002
- @joostdecock
- @KaerMorhan
- @bobgeorgethe3rd
- @BenJamesBen
- @anna-puk 
- @eriese 
- @SeaZeeZee 
- @woutervdub 
- @darigovresearch
- @nchilada
- Lizzomatic
- H0rt3ns14

✨ **Joost**
Proposal to use “Measurement set” instead of “People” on the website so it’s more consistent for the future as users do not always make patterns for humans.

✨ **Ben**
Remove rigid documentation structure for pattern docs. I was thinking about removing the 4-page standard that we use now and maybe add more flexibility and more pages.

_Eriese: I think a lot of information on pages we’re using now is pretty useful to know. I’m up for adding more but I think we should keep the ones we have now._
_Bob: We were discussing the 5th page of “Knowing more” or “Extra information” or “Further reading”._
_Joost: I think Ben was talking more about flexibility to allow people to add more pages if it’s needed and not adding another standard page. Because now it’s not technically impossible but also is not encouraged. Please correct me if I’m wrong. The only thing that I’m worrying about is possible translation mistakes about the structure._
_Karen: I have a question: the original 4 pages, do we want to keep this structure or should it be more of a guideline and we give users complete freedom of structuring their documentation?_
_Ben: I was thinking about making it more of a style guide and then giving the freedom._
_Joost: I don’t want to end up in a situation where the answer to the “Where can I find instructions for this pattern?” is “Well, it depends on which pattern you’re talking about”. We’re not married to the current structure, we don’t need to decide it know, we can leave it for more discussion, maybe we will come up with a more sufficient structure for documentation._
_Ben: I don’t mind that, I think it’s hard to talk about changing something about this without knowing the final result._

✨ **Eriese**
Hide things differently.
Now hiding is a bit convoluted. I am now working on the layout and I need for some things to stay hidden. I’ve come up with 2 convoluted proposals on how to change hiding things. The idea is to be able to hide parts independently. For example, parts that are derived from another parts, that are hidden don’t hide the whole branch of parts that are depending on each other.

_Joost: Yes, we have 2 types of dependences: from and after. “From” is using options of parts and “after” uses the whole part and build on top of that. “Nested” describes dependency on something that already has a dependency on something before that. I propose to change the name of “nested” as it doesn’t feel intuitive. For now, we have 5 different groups of parts and we have only 3 ways of hiding them, which doesn’t have the flexibility we need. We want to be able to completely override everything for the parts so it doesn’t matter what dependences there are._
_Eriese: We need to find a solution to hide previous dependent parts without going the whole way back and changing it there and then changing it further down the branch. Feedback from designers would be very welcome in this discussion._
_Wouter: I think, from my perspective, I can live with everything as far as it’s documented. Should we think about parts that are not meant to be drafted? Something that is only done for calculations._
_Eriese: Hide self should take care of that._

Proposals for names of dependent parts: ancestors, inherited, offshoots.

✨ **Eriese**
Display cutting instructions on parts.
Again, I’m working on layout and cutting lists. I find it useful to have some of the instructions printed directly on the pattern instead of just having cutting instructions. I’ve ran into a problem of macro titles not being available to implement some of the cutting instructions.

_Ben: I think it would be useful to add a separate micro cutting._
_Wouter: Could it be a part of “part config”?_
_Eriese: The reason we have decided not to put it into the config is because it varies from pattern to pattern and there are a lot of changes based on configuration. I have also implemented a way to add cutting instructions on more complicated patterns._
_Joost: I think it’s a good idea to print instructions on the pattern. I’m in favor of making it a part of a macro title. The only thing that we need to keep in mind is that it needs to be translatable. Designer will need to call the title macro for patterns which are dependent on another pattern. The title needs to be overwritten so the cutting list is updated._
_Wouter: What will happen to the parts that don’t have a title macro?_
_Eriese: Printing instructions will not be printed on them._
_Joost: And information about cutting those parts will be in the documentation._

✨ **Joost**
The fuck trees setting.
_Joost: I think we need a more socially acceptable name._
_Eriese: Save space. Fuck trees would be “save space=false”. This setting would be used during drafting to see if shortened pattern pieces should be drawn at full size, and if parts drawn from other parts (e.g. facings) should be drafted separately. My thinking is that this option will be helpful for cutting layout, where we need every piece in full in order to determine fabric usage._

✨ **Bob**
I have found a bug about titles. I will create an issue. #3594 

✨ **Karen**
Daylight savings are coming so heads up for that!

**Comment by joostdecock** [comment]
Closing to increase visibility of active discussions
